processor/src/main/java/org/mapstruct/ap/internal/model/EnumMappingMethod.java

```diff
@@ -149,7 +149,8 @@ else if ( !sourceEnumConstants.contains( mappedConstant.getSourceName() ) ) {
                         );
                         foundIncorrectMapping = true;
                     }
-                    else if ( !targetEnumConstants.contains( mappedConstant.getTargetName() ) ) {
+                    else if ( !targetEnumConstants.contains( mappedConstant.getTargetName() )
```

Comment:
Thinking about it, I'd say this should work the other way around, too. I.e. "null" on the source side should be mapped into a given value of the target side enum.